core/arch/arm/kernel/secure_partition.c

```diff
@@ -68,6 +85,32 @@ struct sp_session *sp_get_session(uint32_t session_id)
 	return NULL;
 }
 
+TEE_Result sp_get_partitions_info(struct ffa_partition_info *fpi,
+				  size_t *elements)
+{
+	size_t in_elements = *elements;
+	struct sp_session *s = NULL;
+	uint32_t count = 0;
+
+	TAILQ_FOREACH(s, &open_sp_sessions, link) {
+		if (count < in_elements) {
```

Comment:
This block can be simplified as:
``` C
if (s->state == sp_dead)
        continue;
if (count < in_elements)
         spmc_fill_partition_entry(fpi + count, s->endpoint_id);
count++

Reply:
I believe that `count` should be increased after the call to `spmc_fill_partition_entry()`